torat_client/rpcapi.go

```diff
@@ -211,6 +211,18 @@ func (a *API) GetHardware(v shared.Void, r *shared.Hardware) error {
 
 	return nil
 }
+
+// GetOSInfo information for a client
+func (a *API) GetOSInfo(v shared.Void, r *shared.OSInfo) error {
+	release := osinfo.GetVersion()
+	r.Runtime = release.Runtime
+	r.OSArch = release.Arch
+	r.OSName = release.Name
+	r.OSVersion = release.Version
+	return nil
+}
```

Comment:
What is your reasoning for adding this? Doesn't GetHardware do sth. very similar?

Reply:
HardwareInfo crashes on Windows, this way atleast you can get a reliable Operating System Information.

check slack